### DISCUSSION TITLE
ToListAsync with Select

### DISCUSSION THREAD

**Discussion by mihail-nikolov** [ramantsitou/MockQueryable#72] [Q&A] [answered]
hello, I have a select, before calling ToListAsync(). This results in NullReferenceException, static members and non public members are NULL.

example:

var res = _repository.GetAll()
.Select(p => new UserDto
{
Id = p.Id,
Name = p.Name
})
.Where(x => includeArchived ? true : !x.IsArchived)
.ToListAsync();

in UT:
var mockedQueryable = DummyFactory.Users().BuildMock();
_repository.Setup(x => x.GetAll()).Returns(mockedQueryable);

Am I missing something?

**Comment by ramantsitou** [comment, marked as answer]
@mihail-nikolov  did you try to compare your test case with [provided examples](https://github.com/romantitov/MockQueryable/blob/master/src/MockQueryable/MockQueryable.Sample/MyServiceMoqTests.cs)? There are some similar cases which works fine. I believe if you compare these examples with your test case and fine difference - you will be able to find an answer to your question by yourself.
